pycyphal/transport/udp/_ip/_endpoint_mapping.py

```diff
@@ -4,214 +4,232 @@
-        mask = 2**ipaddress.IPV6LENGTH - 1
-        ty = ipaddress.IPv6Address
+        msb = MULTICAST_PREFIX | SNM_BIT_MASK  
     else:
-        assert False
-    if local_ip_address.is_multicast:
-        raise ValueError(f"The local address shall be a unicast address, not multicast: {local_ip_address}")
-    return ty((int(local_ip_address) & (mask ^ IP_ADDRESS_NODE_ID_MASK)) | node_id)  # type: ignore
+        raise NotImplementedError("IPv6 is not yet supported; please, submit patches!")
+    return ty(msb | destination_node_id)  # type: ignore
 
 
-def unicast_ip_to_node_id(local_ip_address: IPAddress, node_ip_address: IPAddress) -> typing.Optional[int]:
+def service_multicast_group_to_node_id(multicast_group: IPAddress) -> typing.Optional[int]:
     """
-    Returns the node-ID if the node IP address and the local IP address belong to the same subnet.
-    Returns None if the node is not a member of the local subnet.
-    Raises a value error if either address is a multicast group address.
+    The inverse of :func:`service_node_id_to_multicast_group`.
+    The return value is None if:
+    - is a broadcast multicast group, or
+    - the multicast group is not valid per the current Cyphal/UDP specification.
 
     >>> from ipaddress import ip_address
-    >>> unicast_ip_to_node_id(ip_address('127.42.1.1'), ip_address('127.42.1.200'))
+    >>> service_multicast_group_to_node_id(13, ip_address('239.1.0.123'))
+    123
+    >>> service_multicast_group_to_node_id(13, ip_address('239.1.1.200'))
     456
-    >>> unicast_ip_to_node_id(ip_address('127.0.0.99'), ip_address('127.0.0.99'))
-    99
-    >>> unicast_ip_to_node_id(ip_address('127.99.1.1'), ip_address('127.42.1.200'))  # Returns None
-    >>> unicast_ip_to_node_id(ip_address('239.42.1.1'), ip_address('127.42.1.200'))
-    Traceback (most recent call last):
-      ...
-    ValueError: Multicast group address cannot be a local IP address...
-    >>> unicast_ip_to_node_id(ip_address('127.42.1.1'), ip_address('239.42.1.200'))
-    Traceback (most recent call last):
-      ...
-    ValueError: Multicast group address cannot be mapped to a node-ID...
+    >>> service_multicast_group_to_node_id(13, ip_address('239.52.1.200')) # -> None (broadcast)
     """
-    if local_ip_address.is_multicast:
-        raise ValueError(f"Multicast group address cannot be a local IP address: {local_ip_address}")
-    if node_ip_address.is_multicast:
-        raise ValueError(f"Multicast group address cannot be mapped to a node-ID: {node_ip_address}")
-    if (int(local_ip_address) | IP_ADDRESS_NODE_ID_MASK) == (int(node_ip_address) | IP_ADDRESS_NODE_ID_MASK):
-        return int(node_ip_address) & IP_ADDRESS_NODE_ID_MASK
+
+    candidate = int(multicast_group) & DESTINATION_NODE_ID_MASK
+    if candidate == DESTINATION_NODE_ID_MASK:
+        candidate = None
+    if service_node_id_to_multicast_group(candidate) == multicast_group:
+        return candidate
```

Comment:
I think this should be simplified: just check if the masked value is 0xFFFF, return None if so, otherwise return the value. The old logic is somewhat convoluted to support different domain-IDs, which I presume we are going to drop for now (per the discussion on the forum and judging by the lack of its presence here).

Reply:
No checking if the multicast group is valid per Cyphal/UDP spec? (For example if SNM bit is not set)